admin/js/form-script.js

```diff
@@ -1,9 +1,144 @@
-function toggleForm( formID, hide ){
-	if( formID == 'new_theme_metadata_form' && !hide ) {
-		toggleForm( 'new_variation_metadata_form', true );
+// Toggles the visibility of the forms based on the selected theme type
+// eslint-disable-next-line no-unused-vars
+function toggleForm( element ) {
+	if ( ! element?.value ) return;
+	const themeType = element.value;
+	hideAllForms();
+
+	switch ( themeType ) {
+		case 'export':
+		case 'save':
+			// Forms should stay hidden
+			// Default submit button should be shown
+			hideDefaultSubmitButton();
+			break;
+
+		case 'child':
+		case 'clone':
+		case 'blank':
+			// Show New Theme form
+			document
+				.getElementById( 'new_theme_metadata_form' )
+				.toggleAttribute( 'hidden', false );
+
+			hideDefaultSubmitButton( true );
+			resetThemeTags( element.value );
+			validateThemeTags( 'subject' );
+			break;
+
+		case 'variation':
+			// Show Variation form
+			document
+				.getElementById( 'new_variation_metadata_form' )
+				.toggleAttribute( 'hidden', false );
+
+			hideDefaultSubmitButton();
+			break;
+
+		default:
+			break;
 	}
-	if( formID == 'new_variation_metadata_form' && !hide ) {
-		toggleForm( 'new_theme_metadata_form', true );
+}
+
+function hideAllForms() {
+	const allForms = document.querySelectorAll( '.theme-form' );
+	allForms.forEach( ( form ) => {
+		form.toggleAttribute( 'hidden', true );
+	} );
+}
+
+// Hide the default submit button if a long form is being displayed
+function hideDefaultSubmitButton( shouldHide ) {
+	const defaultSubmitButton = document.querySelector(
+		'#col-left input[type="submit"]'
+	);
+
+	if ( ! shouldHide ) {
+		defaultSubmitButton.style.display = 'block';
+	}
+	if ( shouldHide ) {
+		defaultSubmitButton.style.display = 'none';
+	}
+}
+
+// Handle theme tag validation
+function validateThemeTags( tagCategory ) {
+	if ( ! tagCategory ) return;
+	let checkboxes;
+
+	if ( 'subject' === tagCategory ) {
+		checkboxes = 'input[name="theme[tags-subject][]"]';
+	}
+
+	limitCheckboxSelection( checkboxes, 3 );
+}
+
+// Takes a checkbox selector and limits the number of checkboxes that can be selected
+function limitCheckboxSelection( checkboxesSelector, max = 0 ) {
+	if ( ! checkboxesSelector ) return;
+
+	const allCheckboxes = document.querySelectorAll( checkboxesSelector );
+	const checked = document.querySelectorAll(
+		`${ checkboxesSelector }:checked`
+	);
+	const unchecked = document.querySelectorAll(
+		`${ checkboxesSelector }:not(:checked)`
+	);
+
+	if ( checked.length >= max ) {
+		for ( let i = 0; i < unchecked.length; i++ ) {
+			unchecked[ i ].setAttribute( 'disabled', true );
+		}
+	}
+
+	if ( checked.length < max ) {
```

Comment:
Could this be an `else` statement of the conditional right before this one?

Reply:
Yeah, I think that works the same!